### PR TITLE
Incorrect startPosition of a SingleVariableDeclaration in the catch block

### PR DESCRIPTION
**Pull request by subyssurendran666**
## What it does
https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2564

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

File: org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/CommentRecorderParser.java [hunk 1]
```diff
@@ -203,6 +203,11 @@ public int flushCommentsDefinedPriorTo(int position) {
 		return position;
 	}
 
+	@Override
+	protected void consumeExitTryBlock() {
+		flushCommentsDefinedPriorTo(this.scanner.currentPosition);
```
Comment (jarthana) [inline org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/CommentRecorderParser.java:200]: @subyssurendran666 See some of the other overridden consume* methods in this class that call the equivalent methods on super after doing what they are doing. I think we should do the same here as well.

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverter18Test.java [hunk 2]
```diff
@@ -5483,4 +5483,36 @@ interface I {
...
+	List<ASTNode> statements = block.statements();
+	TryStatement tryStatement = (TryStatement) statements.get(0);
+	List<ASTNode> catchClauseList = tryStatement.catchClauses();
+	CatchClause catchClause = (CatchClause) catchClauseList.get(0);
+	SingleVariableDeclaration svd = catchClause.getException();
+
+	assertEquals("Not a Single Variable Declaration", ASTNode.SINGLE_VARIABLE_DECLARATION, svd.getNodeType());
+	assertEquals("Not a Simple Type", ASTNode.SIMPLE_TYPE, svd.getType().getNodeType());
+	assertEquals("Not a Simple Name", ASTNode.SIMPLE_NAME, svd.getName().getNodeType());
+	assertEquals("Single Variable Declaration length is not correct", svd.getLength(), 18);
```
Comment (jarthana) [inline org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverter18Test.java:5514]: For anyone verifying this change or even those touching this testcase in future, it's easier if we use actual content to get the expected source range. For e.g "RuntimeException e".length(), contents.indexOf() etc.

**Review by jarthana — CHANGES_REQUESTED** [review]

File: org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/CommentRecorderParser.java (hunk 1 above)
Comment (subyssurendran666) [reply org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/CommentRecorderParser.java]: Done

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverter18Test.java (hunk 2 above)
Comment (subyssurendran666) [reply org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverter18Test.java]: Done

**Comment by jarthana** [comment]
@subyssurendran666 Sorry for input in a piecemeal manner. Can you also add a testcase with a comment on the catch parameter? Like this:
`catch(/** abc*/ Exception e) {}`
We should test that the source positions includes the comment.

**Comment by subyssurendran666** [comment]
> @subyssurendran666 Sorry for input in a piecemeal manner. Can you also add a testcase with a comment on the catch parameter? Like this: `catch(/** abc*/ Exception e) {}` We should test that the source positions includes the comment.

@jarthana I have added a new test case. Could you please review my changes ?

**Review by jarthana — APPROVED** [review]

**Pull request merged**